### ISSUE TITLE
Workflow containing the meme tool cannot be saved

### ISSUE THREAD

**Issue by gregvonkuster** [galaxyproject/galaxy#1877]
Any workflow that includes the meme tool from here https://toolshed.g2.bx.psu.edu/view/iuc/meme_meme/798018a73e3f cannot be save.  After clicking the save button, the workflow editor hangs forever.  This issue exists in dev and goes back to at least release 15.10.  Could the fix please be ported back to release 15.10?  Thanks!

![no_save](https://cloud.githubusercontent.com/assets/6947053/13570759/3f6b73f0-e43e-11e5-9dab-ad05a8ceae50.png)

**Comment by mvdbeek** [comment]
I can reproduce this and it seems to be a problem with bibtex again;
I got this in the javascript console:

```
jquery.migrate.js:21JQMIGRATE: Logging is active
create:244 bibtex? undefined
jquery.js:9625Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
citation-model.js:25Uncaught TypeError: Cannot read property 'Fields' of undefined
```

**Comment by mvdbeek** [comment]
maybe @carlfeberhard has an idea what do here?

**Comment by carlfeberhard** [comment]
@gregvonkuster It looks like you're using `@published` for your bibtex entry type:
https://github.com/galaxyproject/tools-iuc/blob/master/tools/meme/meme.xml#L326

That's an invalid type:
https://en.wikipedia.org/wiki/BibTeX#Entry_types
https://github.com/galaxyproject/bib2json/blob/master/Parser.js#L79

And the parser is throwing `.errors` instead of `.entries`. I believe you want to use `@article` instead.

Apart from that, an invalid bibtex shouldn't throw up errors and prevent a user from saving a workflow - especially since the bibtex isn't something they can control. I'll make a '[15.10]' fix to sidestep the error that's stopping the workflow save. This should make the model a bit more bullet proof (but empty when invalid). I'll output any parsing errors to the console for better tracking.

**Comment by bgruening** [comment]
@gregvonkuster can you fix the bibtex of the meme tool please.
Thanks @carlfeberhard!

Fixed in #1884

**Comment by gregvonkuster** [comment]
PR is here: https://github.com/galaxyproject/tools-iuc/pull/662 Thanks for the help on this!  ;)
